File: Mago.java
class Mago extends Personagem{
    protected  int mana;

    public Mago(String nome, int vida, int nivel, int mana){
        super(nome, vida, nivel);
        this.mana = mana;
    }

    //Getters
    public int getMana(){
        return(mana);
    }

    //Setters
    public void setMana(int mana){
        if (mana >= 0){
            this.mana = mana;
        } else{
            System.out.printf("%nInsira um valor válido para mana%n");
        }
    }
    
    @Override
    public void ficha(){
        super.ficha();
        System.out.printf("%nMana: %d%n", mana);
    }
    
}

File: Polimorfismo - Batalha Final/Python/Mago.py
from Personagem import Personagem

class Mago(Personagem):
    mana = 0

    def __init__(self, nome, vida, nivel, mana):
        super().__init__(nome, vida, nivel)
        self.mana = mana

    # Getter
    def getMana(self):
        return self.mana

    # Setter
    def setMana(self, mana):
        if mana >= 0:
            self.mana = mana
        else:
            print("\nInsira um valor válido para mana")

    # Override
    def ficha(self):
        super().ficha()
        print(f"\nMana: {self.mana}")

    def habilidade(self):
        return self.nome + "usa Rajada Arcana"